decoder.rb

```diff
@@ -0,0 +1,54 @@
+DECODER_HASH = {
+  '.-' => 'A',
+  '-...' => 'B',
+  '-.-.' => 'C',
+  '-..' => 'D',
+  '.' => 'E',
+  '..-.' => 'F',
+  '--.' => 'G',
+  '....' => 'H',
+  '..' => 'I',
+  '.---' => 'J',
+  '-.-' => 'K',
+  '.-..' => 'L',
+  '--' => 'M',
+  '-.' => 'N',
+  '---' => 'O',
+  '.--.' => 'P',
+  '--.-' => 'Q',
+  '.-.' => 'R',
+  '...' => 'S',
+  '-' => 'T',
+  '..-' => 'U',
+  '...-' => 'V',
+  '.--' => 'W',
+  '-..-' => 'X',
+  '-.--' => 'Y',
+  '--..' => 'Z',
+  '.----' => '1',
+  '..---' => '2',
+  '...--' => '3',
+  '....-' => '4',
+  '.....' => '5',
+  '-....' => '6',
+  '--...' => '7',
+  '---..' => '8',
+  '----.' => '9',
+  '-----' => '0'
+}.freeze
+
+def decode(str)
+  words = str.split('   ')
+  decoded = ''
+  words.each do |word|
+    decoded_word = ''
+    word.split.each do |char|
+      decoded_word += DECODER_HASH[char]
+    end
+    decoded += " #{decoded_word}"
+  end
+
+  decoded.strip
+end
```

Comment:
- [ ] Greate work @jeanbuhendwa and @adamilare :+1: but please from the project requirement, you should create three methods:point_down: 
  
  - a method to decode a Morse code character, takes a string parameter and returns the corresponding character in uppercase (e.g. `decode_char(".-")` returns `"A"`). :pray: 
   - a method to decode an entire word in Morse code, takes a string parameter, and returns the string representation. Every character in a word will be separated by a single space (e.g.`decode_word("-- -.--")` returns `"MY"`). :pray: 
  
  - a method to decode the entire message in Morse code, takes a string parameter and returns the string representation. Every word will be separated by 3 spaces (e.g.`decode("-- -.--   -. .- -- .")` returns `"MY NAME"`). :pray: